### DISCUSSION TITLE
ESPHome config for EP3246 ?

### DISCUSSION THREAD

**Discussion by p1ranha** [TillFleisch/ESPHome-Philips-Smart-Coffee#82] [General]
Hi,

is there an actual full ESPHome config/code available for EP3246 ?

Can someone share?

I tried, but the documentation is not really clear on many things: for example what is difference between SELECT_ESPRESSO and MAKE_ESPERSSO. How to make double espresso? How to enable already grinded coffee (normally it's long press on size)? Do we get status on missing water of coffee pucks container full?

Regards
Robert

**Comment by quack3d** [comment]
Here is mine for EP3243: https://dpaste.org/SwbyC

SELECT_ESPRESSO is like pressing espresso one time on the display. MAKE_ESPRESSO selects AND starts brewing espresso. I don't use any of the MAKE_ commands. I use automations in HA that SELECT_COFFEE, pause 2 seconds, select coffe strength, pause 2 seconds, select size, pause 2 seconds and then hit PLAY/PAUSE. Those automations can also be done with the lambda in ESPHome but I prefer to do it in HA.

For a double espresso I'd make an automation that presses twice on SELECT_ESPRESSO. 

For pre-ground coffee you can see my 'Select Ground Coffee' above. You just use long_press: true.

You get status of empty water and container full with the Status-sensor, yes.

**Comment by TillFleisch** [comment]
> is there an actual full ESPHome config/code available for EP3246 ?

You should be able to use the default example configuration [example.yaml](https://github.com/TillFleisch/ESPHome-Philips-Smart-Coffee/blob/main/example.yaml) with the only required change being `model: EP_3243` under the `philips_coffee_machine` component.

> what is difference between SELECT_ESPRESSO and MAKE_ESPERSSO

The `select` option only presses the select button (for the respective beverage) and the `make` options additionally also presses the play/pause to prepare the beverage. Essentially a shortcut to eliminate the need for a separate script/automation in ESPHome.

> How to make double espresso?

This can be achieved by using the `select` option twice, like you would do manually. You can automate this with a script, but there isn't a more elegant way, as this component can only imitate user interactions.

> How to enable already grinded coffee (normally it's long press on size)?

Same as above, except buttons have a `long_press` options which will cause them to imitate a long press, which can be used to select ground coffee.

> Do we get status on missing water of coffee pucks container full?

Yes. Currently only via the text status sensor, though a template sensor could be used to create separate binary sensors for each if needed.

Damn @quack3d  was faster, thank you!

**Comment by p1ranha** [comment]
Awesome! Thank you for the prompt answer! I am waiting for my PCB to arrive, once I get it and tweak the code, I will make sure to share my yaml and card configuration so you can put it in examples directory for future users.

**Reply by TillFleisch** [reply]
[Show&Tell](https://github.com/TillFleisch/ESPHome-Philips-Smart-Coffee/discussions/categories/show-and-tell) would a good place for the (I presume HA-Frontend) card configuration, as it's only distantly related to this repo and user/taste specific. We can point interested users there from the readme.

**Comment by p1ranha** [comment]
To be honest having a complete config with all the features that just need to be removed helps a lot for beginners (and there are lots of them lurking here) :)

**Reply by TillFleisch** [reply]
I have no problem with the addition of an example folder, which contains examples for different configurations/machines. In fact, that is probably a good idea for beginners and should have been done a long time ago. There is already one for the EP2220 [here](https://github.com/TillFleisch/ESPHome-Philips-Smart-Coffee/blob/main/templates/Philips_EP2220.yaml), which can be used for factory firmware generation. This is not pointed out anywhere for a good reason.
I don't want to cause confusion by pointing to a "too complex/non-standard" configuration. Furthermore, I don't want anyone to end up with an "insecure" setup, that's why all of the given examples are partial and omit the user-specific preamble (wifi credentials and so on).
I also don't want to bombard HA-instances with "useless/unused" entities, hence the current example is quite small, but I can see that a more complete config may be useful for newcomers even if it's just used as a guide.

Regarding the Show&Tell section: I was only referring to the card configuration, as that is something HA/user-specific and others may have other setups.
